spock-specs/mock-integration/src/test/groovy/MockingIntegrationSpec.groovy

```diff
@@ -41,6 +49,10 @@ class MockingIntegrationSpec extends Specification {
   }
 
   @IgnoreIf({ TEST_TYPE == "plain" })
+  @IgnoreIf(
+    value = { jvm.java21Compatible && TEST_TYPE.contains("cglib") },
+    reason = "Cglib doesn't support running on Java 21+"
+  )
```

Comment:
Creation of a local meta annotation could be considered to avoid duplication

Reply:
We don't really have a good place to share across the projects it at the moment